iml-agent/src/action_plugins/stratagem/action_filesync.rs

```diff
@@ -5,33 +5,54 @@
 use crate::{agent_error::ImlAgentError, env, lustre::search_rootpath};
 use futures::{future::try_join_all, future::Future, stream, StreamExt, TryStreamExt};
 use iml_wire_types::{FidError, FidItem};
-use liblustreapi::LlapiFid;
 use std::path::PathBuf;
 use structopt::clap::arg_enum;
 use tokio::{fs, process::Command};
 
+arg_enum! {
+    #[derive(Debug, serde::Deserialize, Clone, Copy, PartialEq)]
+    pub enum ActionType {
+    Push,
+    Pull,
+    }
+
+}
 struct Work {
+    file_path: PathBuf,
     fid: FidItem,
-    src_file: String,
-    dest_file: String,
 }
 
 const LARGEFILE: u64 = 1024 * 1024 * 1024;
 
 fn do_rsync<'a>(
+    src_root: &'a PathBuf,
+    dest_root: &'a PathBuf,
     work_list: &'a [Work],
 ) -> impl Future<Output = Result<Vec<FidError>, ImlAgentError>> + 'a {
     stream::iter(work_list)
-        .map(|work| async move {
+        .map(move |work| async move {
```

Comment:
```suggestion
        .map(
            move |Work {
                      file_path,
                      fid: FidItem { fid, data },
                  }| async move {
```

Once `work_list` is owned, the data can be destructured to own all the pieces.

Reply:
Done